src/bidi.rs

```diff
@@ -658,9 +655,8 @@ where
 {
     let mut max_level = 0;
     let mut lowest_odd_level = 255;
-    let mut idx = 0;
     let len = order.len();
-    for i in 0..len {
+    for (idx, i) in (0..len).enumerate() {
```

Comment:
This seems weird. `i` and `idx` are always the same.

Reply:
They are indeed. Simply using `i` in place of `idx` is logically correct, but doesn't satisfy Clippy. ☹️ The change here was the smallest change that would satisfy, and I thought that it isn't the worst even if redundant.

Reply:
Alright I rewrote this as:

```rust
for (i, o) in order.iter_mut().enumerate() {
    *o = i;
```